terraform/infrastructure/openstack/main.tf

```
@@ -176,44 +188,38 @@ resource "openstack_networking_secgroup_rule_v2" "nodeport_udp" {
 }
 
 resource "openstack_networking_secgroup_rule_v2" "tcp_port_forward" {
-  for_each = toset(flatten([
-    local.ports_kubernetes,
-    local.ports_bootstrapper,
-    local.ports_konnectivity,
-    local.ports_verify,
-    local.ports_recovery,
-    var.debug ? [local.ports_debugd] : [],
-  ]))
+  for_each          = local.control_plane_named_ports
```

```suggestion
  for_each          = {for i,v in local.control_plane_named_ports: i=>v}
```

In my test, this was required